pypeit/slittrace.py

```diff
@@ -258,9 +261,14 @@ def _validate(self):
         if self.slitbitm is None:
             self.slitbitm = ','.join(list(self.bitmask.keys()))
         else:
-            # Validate
-            if self.slitbitm != ','.join(list(self.bitmask.keys())):
-                msgs.error("Input BITMASK keys differ from current data model!")
+            # Validate -- All of the keys must be present and in current order, but new ones can exist
+            bitms = self.slitbitm.split(',')
+            curbitm = list(self.bitmask.keys())
+            for kk, bit in enumerate(bitms):
+                if curbitm[kk] != bit:
+                    msgs.error("Input BITMASK keys differ from current data model!")
+            # Update to current, no matter what
+            self.slitbitm = ','.join(list(self.bitmask.keys()))
```

Comment:
This should get updated once we propagate BitMaskArray usage.  These are DataContainer objects that do version checking.  We could update it to this kind if detailed bit checking, too.

Reply:
sounds good